distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso

```diff
@@ -110,48 +120,632 @@ type Array
+    contains self elem = self.any ix-> ix == elem
+
+    ## Performs a pair-wise operation passed in `function` on consecutive
+       elements of `self` and `that`.
+
+       Arguments:
+       - that: The array to zip with `self`.
+       - function: The function used to combine pairwise elements of `self` and
+         `that`.
+
+       The result of this function is a `Vector` of length being the shorter of
+       `self` and `that`, containing results of calling `function`.
+
+       > Example
+         Calculate the pairwise sum of two arrays.
+
+             [1, 2, 3].to_array.zip [4, 5, 6].to_array (+)
+
+       > Example
+         When the `function` is not provided, it defaults to creating a pair
+         of both elements.
+
+             [1, 2, 3].to_array.zip [4, 5, 6].to_array == [[1, 4].to_array, [2, 5].to_array, [3, 6].to_array].to_array
+    zip : Vector Any -> (Any -> Any -> Any) -> Vector Any
+    zip self that function=[_,_] = Vector.zip self that function
+
+    ## UNSTABLE
+       Returns the array as a `Vector`.
+    to_vector : Vector
+    to_vector self = Vector.from_polyglot_array self
+
+    ## Reverses the array, returning a `Vector` with the same elements, but in
+       the opposite order.
+
+       > Example
+         Reverse a two-element array.
+
+             [1, 2].to_array.reverse
+    reverse : Vector Any
+    reverse self = Vector.reverse self
+
+    ## Applies a function to each element of the array.
+
+       Unlike `map`, this method does not return the individual results,
+       therefore it is only useful for side-effecting computations.
+
+       Arguments:
+       - function: A function to apply to each element of the array.
+
+       > Example
+         Print each element in the array to standard output.
+
+             [1, 2, 3, 4, 5].to_array . each IO.println
+    each : (Any -> Any) -> Nothing
+    each self f = Vector.each self f
+
+    ## Applies a function to each element of the array.
+
+       Arguments:
+       - function: A function to apply that takes an index and an item.
+
+       The function is called with both the element index as well as the
+       element itself.
+
+       Unlike `map`, this method does not return the individual results,
+       therefore it is only useful for side-effecting computations.
+
+       > Example
+         Print each element in the array to standard output.
+
+             [1, 2, 3, 4, 5].to_array . each_with_index (ix->elem-> IO.println Pair ix elem)
+    each_with_index : (Integer -> Any -> Any) -> Nothing
+    each_with_index self f = Vector.each_with_index self f
+
+    ## Concatenates two arrays, resulting in a new `Vector`, containing all the
+       elements of `self`, followed by all the elements of `that`.
+
+       Arguments:
+       - that: The array to concatenate to the end of `self`.
+
+       > Example
+         Concatenate two single-element arrays.
+
+             [1].to_array + [2].to_array
+    + : Vector Any -> Vector Any
```

Comment:
The type should be an Array?

Reply:
The in type can be either `Vector` or `Array`. The output type is `Vector`.